src/NHibernate/Linq/QuerySourceNamer.cs

```diff
@@ -15,11 +15,16 @@ public class QuerySourceNamer
 		private int _differentiator = 1;
 
 		public void Add(IQuerySource querySource)
+		{
+			Add(querySource, CreateUniqueName(querySource.ItemName));
```

Comment:
This creates unused unique names when the `_map` already contains the `querySource`. It looks to me as a bad move.
I think this method should not call its overload but do all its little job by itself, allowing to keep an effective shortcut without evaluating twice the shortcut.

Reply:
I overlooked that, fixed.